nncf/quantization/algorithms/min_max/algorithm.py

```diff
@@ -59,81 +62,103 @@ class MinMaxQuantizationParameters(AlgorithmParameters):
+        :param activation_granularity: Type of quantization granularity for activation quantizers.
+        Could be per-channel or per-tensor.
+        :param activation_signedness_to_force: Defines whether the datatype of the activation quantizers
+         should be forced. True if the quantizer *must* be signed, False if *must* be unsigned,
+        None if the signed/unsigned attribute should be determined based on the incoming activation
+        statistics during range initialization.
+        activation_signedness_to_force
         :param target_device: Target device for the settings of the quantization pipeline.
-        :param range_type: Range types for the quantizer's parameters.
+        :param range_type: Type of statistics range calculation.
         :param quantize_outputs: Boolean value that says whether quantize outputs or not.
-        :param ignored_scopes: List of the layers that should not quantized during the process.
+        :param ignored_scopes: List of the layers which input must not be quantized.
         """
-        weight_mode, activation_mode = self._determine_weight_activation_modes(preset)
-        self.weight_quantizer_config = self._determine_quantizer_config(weight_bits, weight_granularity, weight_mode)
-        self.activation_quantizer_config = self._determine_quantizer_config(activation_bits, activation_granularity,
-                                                                            activation_mode)
         self.number_samples = number_samples
         self.target_device = HWConfigType(HW_CONFIG_TYPE_TARGET_DEVICE_MAP[target_device.value])
         self.range_type = range_type
         self.quantize_outputs = quantize_outputs
         self.ignored_scopes = [] if ignored_scopes is None else ignored_scopes
-
-    def to_json(self) -> Dict[str, Union[str, float, int]]:
+        self.global_quantizer_constraints = {}
+        weight_per_channel, activation_per_channel = None, None
+        if weight_granularity is not None:
+            weight_per_channel = weight_granularity == Granularity.PERCHANNEL
+        if activation_granularity is not None:
+            activation_per_channel = activation_granularity == Granularity.PERCHANNEL
+        q_weight_constraints = self._get_quantizer_constraints(QuantizerGroup.WEIGHTS, preset, weight_bits,
+                                                               weight_per_channel, weight_signedness_to_force)
+        q_activation_constraints = self._get_quantizer_constraints(QuantizerGroup.ACTIVATIONS, preset, activation_bits,
+                                                                   activation_per_channel,
+                                                                   activation_signedness_to_force)
+        self.global_quantizer_constraints[QuantizerGroup.WEIGHTS] = q_weight_constraints
+        self.global_quantizer_constraints[QuantizerGroup.ACTIVATIONS] = q_activation_constraints
```

Comment:
Honestly, I don't see much difference whether this is an attribute of `MinMaxQuantization` or `MinMaxQuantizationParameters`. 
I prefer to have the same things in one place. All algorithm parameters are initially processed in `MinMaxQuantizationParameters`. Because of that the `global_quantizer_constraints` is obtained in `MinMaxQuantizationParameter`
If you have any concerns about that, please, let me know

Reply:
Okay, let it be your way.